test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -55,4 +55,42 @@ Describe "Set-Location" -Tags "CI" {
             (Get-Location).Path | Should Be (Get-PSProvider FileSystem).Home
         }
     }
+
+    Context 'Set-Location with last location history' {
+        
+            It 'Should go to last location when specifying minus as a path' {
+                $initialLocation = Get-Location
+                Set-Location ([System.IO.Path]::GetTempPath())
+                Set-Location -
+                (Get-Location).Path | Should Be ($initialLocation).Path
+            }
+
+            It 'Should go to last location when specifying minus as a path using alias' {
+                $initialLocation = Get-Location
+                Set-Location ([System.IO.Path]::GetTempPath())
+                cd -
+                (Get-Location).Path | Should Be ($initialLocation).Path
+            }
+    
+            It 'Should go back to original location when specifying minus as a path twice' {
+                $tempPath = ([System.IO.Path]::GetTempPath())
+                Set-Location $tempLocation
+                $tempLocation = Get-Location
+                Set-Location -
+                Set-Location -
+                (Get-Location).Path | Should Be ($tempLocation).Path
```

Comment:
This test doesn't make sense to me - there is one push, but two pops. How can this pass?

Reply:
I have re-written all tests but the reason why it passed was because `Set-Location` also pushed to the stack (this was a bug that is fixed now)